pkg/database/resync.go

```diff
@@ -4,45 +4,179 @@ package database
-		return err
+		klog.Warningf("Error getting existing resources uids of cluster %s. Error: %+v", clusterName, err)
+	}
+	defer existingRows.Close()
+	for existingRows.Next() {
+		id := ""
+		err := existingRows.Scan(&id)
+		if err != nil {
+			klog.Warningf("Error scanning existing resource row. Error: %+v", err)
+			continue
+		}
+		existingResourcesMap[id] = struct{}{}
+	}
+	logStepDuration(&timer, clusterName, "QUERY existing resources")
+
+	// INSERT or UPDATE resources.
+	// In case of conflict update only if data has changed.
+	for _, resource := range resources {
+		delete(existingResourcesMap, resource.UID)
+		data, _ := json.Marshal(resource.Properties)
+		// TODO: Use goqu to build the query.
+		// TODO: Combine multiple inserts into a single query.
+		queueErr := batch.Queue(batchItem{
+			action: "addResource",
+			query: `INSERT into search.resources as r values($1,$2,$3) ON CONFLICT (uid)
+			DO UPDATE SET data=$3 WHERE r.uid=$1 and r.data IS DISTINCT FROM $3`,
+			uid:  resource.UID,
+			args: []interface{}{resource.UID, clusterName, string(data)},
+		})
+		if queueErr != nil {
+			klog.Warningf("Error queuing resources. Error: %+v", queueErr)
+			return // TODO: return queueErr
+		}
 	}
-	// DELETE from search.edges WHERE cluster=$1
-	delEdgesSql, args, delEdgesSqlErr := goquDelete("edges", "cluster", clusterName)
-	checkError(delEdgesSqlErr, fmt.Sprintf("Error creating query to delete edges for %s.", clusterName))
+	batch.flush()
+
+	// DELETE any previous resources for the cluster that isn't included in the incoming resync event.
+
+	if len(existingResourcesMap) > 0 {
+		// TODO: Use goqu to build the query.
+		resourcesToDelete := make([]string, 0)
+		for resourceUID := range existingResourcesMap {
+			resourcesToDelete = append(resourcesToDelete, "'"+resourceUID+"'")
+		}
+
+		queryStr := fmt.Sprintf("DELETE from search.resources WHERE uid IN (%s)", strings.Join(resourcesToDelete, ","))
 
-	_, err = dao.pool.Exec(ctx, delEdgesSql, args...)
+		deletedRows, err := dao.pool.Exec(ctx, queryStr) // TODO: Use batch.Queue() instead of Exec()
+		if err != nil {
+			klog.Warningf("Error deleting resources during resync of cluster %s. Error: %+v", clusterName, err)
+		}
+		klog.Infof("Deleted %d resources during resync of cluster %s", deletedRows.RowsAffected(), clusterName)
+	}
+	batch.wg.Wait()
+	logStepDuration(&timer, clusterName,
+		fmt.Sprintf("Resync INSERT/UPDATE [%d] DELETE [%d] resources", len(resources), len(existingResourcesMap)))
+}
+
+// Reset Edges
+func (dao *DAO) resyncEdges(ctx context.Context, wg *sync.WaitGroup,
+	edges []model.Edge, clusterName string, syncResponse *model.SyncResponse) {
+	defer wg.Done()
+	timer := time.Now()
+
+	batch := NewBatchWithRetry(ctx, dao, syncResponse)
+	var queueErr error
+
+	// Get all existing edges for the cluster.
+	edgeRow, err := dao.pool.Query(ctx, "SELECT sourceId,edgeType,destId FROM search.edges WHERE cluster=$1", clusterName)
```

Comment:
Should we avoid fetching interCluster edges here, by adding edgetype!='interCluster' , so as not to delete them later? They should be inserted and deleted by the trigger on resources table.